### DISCUSSION TITLE
Execute part of the script only on click/dropdown open?

### DISCUSSION THREAD

**Discussion by stegmatze** [swiftbar/SwiftBar#403] [Q&A] [answered]
I'm using a modified version of the [Octoprint Script](https://github.com/matryer/xbar-plugins/blob/main/Tools/octoprint.1m.sh).

It shows the remaining print time in the Header, that's why it is refreshed every minute.
In the dropdown is a preview of the webcam, which is fetched and converted via imagemagick.

I'm wondering if it's possible to only execute the image part (fetch + convert) only when the dropdown is opened, instead of every time the script runs.

**Comment by melonamin** [comment]
In theory you can write a cleverer plugin, which will have contional blocks depending on the type of refresh trigger. I'm already tracking refresh trigger types in code: 
```
enum PluginRefreshReason {
    case FirstLaunch
    case Schedule
    case MenuAction
    case RefreshAllMenu
    case RefreshAllURLScheme
    case URLScheme
    case Shortcut
    case DebugView
    case NotificationAction
    case PluginSettings
...
``` 

I can expose this as an ENV variable and you will be able to have two branches 1) refresh only header 2) refresh everething.

**Reply by stegmatze** [reply]
An ENV variable would be nice!

**Reply by melonamin** [reply, marked as answer]
Ok, added #404
